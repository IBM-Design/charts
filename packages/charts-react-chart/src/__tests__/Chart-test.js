import React from 'react';
import renderer from 'react-test-renderer';
import Chart from '../';

describe('Chart Component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Chart
        height={500}
        margin={{
          top: 20,
          right: 20,
          bottom: 30,
          left: 50,
        }}
        width={960}
        x={{
          text: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'June', 'July', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'],
        }}
        y={{
          min: 0,
          max: 500,
          tickCount: 5,
        }} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
