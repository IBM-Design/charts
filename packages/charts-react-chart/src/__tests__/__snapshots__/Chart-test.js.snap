exports[`Chart Component should render 1`] = `
<svg
  className={undefined}
  height={500}
  width={960}>
  <g
    transform="translate(50,20)">
    <g
      className="axis axis--x"
      fill="none"
      fontFamily="sans-serif"
      fontSize="10"
      textAnchor="middle"
      transform="">
      <g
        className="tick"
        transform="translate(0, 450)">
        <path
          d="M0,0 L890,0"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.32em"
          fill="#000"
          x={-9}
          y={0.5}>
          0
        </text>
      </g>
      <g
        className="tick"
        transform="translate(0, 360)">
        <path
          d="M0,0 L890,0"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.32em"
          fill="#000"
          x={-9}
          y={0.5}>
          100
        </text>
      </g>
      <g
        className="tick"
        transform="translate(0, 270)">
        <path
          d="M0,0 L890,0"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.32em"
          fill="#000"
          x={-9}
          y={0.5}>
          200
        </text>
      </g>
      <g
        className="tick"
        transform="translate(0, 180)">
        <path
          d="M0,0 L890,0"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.32em"
          fill="#000"
          x={-9}
          y={0.5}>
          300
        </text>
      </g>
      <g
        className="tick"
        transform="translate(0, 90)">
        <path
          d="M0,0 L890,0"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.32em"
          fill="#000"
          x={-9}
          y={0.5}>
          400
        </text>
      </g>
      <g
        className="tick"
        transform="translate(0, 0)">
        <path
          d="M0,0 L890,0"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.32em"
          fill="#000"
          x={-9}
          y={0.5}>
          500
        </text>
      </g>
    </g>
    <g
      className="axis axis--x"
      fill="none"
      fontFamily="sans-serif"
      fontSize="10"
      textAnchor="middle"
      transform="translate(0,450)">
      <g
        className="tick"
        transform="translate(0, 0)">
        <path
          d="M0,0 L0,-450"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.71em"
          fill="#000"
          x={0.5}
          y={9}>
          Jan
        </text>
      </g>
      <g
        className="tick"
        transform="translate(80.9090909090909, 0)">
        <path
          d="M0,0 L0,-450"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.71em"
          fill="#000"
          x={0.5}
          y={9}>
          Feb
        </text>
      </g>
      <g
        className="tick"
        transform="translate(161.8181818181818, 0)">
        <path
          d="M0,0 L0,-450"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.71em"
          fill="#000"
          x={0.5}
          y={9}>
          Mar
        </text>
      </g>
      <g
        className="tick"
        transform="translate(242.72727272727272, 0)">
        <path
          d="M0,0 L0,-450"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.71em"
          fill="#000"
          x={0.5}
          y={9}>
          Apr
        </text>
      </g>
      <g
        className="tick"
        transform="translate(323.6363636363636, 0)">
        <path
          d="M0,0 L0,-450"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.71em"
          fill="#000"
          x={0.5}
          y={9}>
          May
        </text>
      </g>
      <g
        className="tick"
        transform="translate(404.5454545454545, 0)">
        <path
          d="M0,0 L0,-450"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.71em"
          fill="#000"
          x={0.5}
          y={9}>
          June
        </text>
      </g>
      <g
        className="tick"
        transform="translate(485.45454545454544, 0)">
        <path
          d="M0,0 L0,-450"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.71em"
          fill="#000"
          x={0.5}
          y={9}>
          July
        </text>
      </g>
      <g
        className="tick"
        transform="translate(566.3636363636364, 0)">
        <path
          d="M0,0 L0,-450"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.71em"
          fill="#000"
          x={0.5}
          y={9}>
          Aug
        </text>
      </g>
      <g
        className="tick"
        transform="translate(647.2727272727273, 0)">
        <path
          d="M0,0 L0,-450"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.71em"
          fill="#000"
          x={0.5}
          y={9}>
          Sep
        </text>
      </g>
      <g
        className="tick"
        transform="translate(728.1818181818181, 0)">
        <path
          d="M0,0 L0,-450"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.71em"
          fill="#000"
          x={0.5}
          y={9}>
          Oct
        </text>
      </g>
      <g
        className="tick"
        transform="translate(809.090909090909, 0)">
        <path
          d="M0,0 L0,-450"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.71em"
          fill="#000"
          x={0.5}
          y={9}>
          Nov
        </text>
      </g>
      <g
        className="tick"
        transform="translate(890, 0)">
        <path
          d="M0,0 L0,-450"
          fill="none"
          stroke="#CDDBF1"
          strokeWidth="1" />
        <text
          dy="0.71em"
          fill="#000"
          x={0.5}
          y={9}>
          Dec
        </text>
      </g>
    </g>
  </g>
</svg>
`;
